### ISSUE TITLE
About Training Procedure

### ISSUE THREAD

**Issue by Le0000000000n** [rpautrat/SuperPoint#76]
Hi! Thanks for your great work. 

I don't know if I ever understand it correctly. Based on my understanding, the training procedure can be divided into 3 parts:

1. use synsetic dataset to train a detector (magic_point_base).
2. use magic_point_base to generate pseudo ground truth labels for real world data. use homographic adaptation to augment the data. Train another detector (magic_point_1, trained from random initial weights rather than from magic_point_base weights) on this augmented dataset with pseudo ground truth labels. repeat this step several times until magic_points_x is able to perform well in real world data.
3. use magic_points_x to generate pseudo ground truth labels for real world data. use homographic adaptation to augment the data. Train a pair of descripter and detector (super_point_1) on this augmented dataset with pseudo ground truth labels.

My questions are:

1. when we train magic point model a second round, are we fine-tuning the base model (magic_point_base) or we training a new model from scratch?
2. it seems that detector (magic point) is trained several round while detector + descriptor (super point) is trained only once. Does it help if I repeat training super point extra rounds like magic point does (generate labels with old version model, train new version model on those labels)? 

Thanks!

**Comment by rpautrat** [comment]
Hi!

Yes, the training procedure is exactly this. Regarding your questions:

1. In this repo we train a new model from scratch when starting a new round. I don't know if it was also the case for the original implementation of Magic Leap though.

2. The extra rounds are mainly to improve the quality of the pseudo ground truth feature points. If the quality of the points at the end of step 2 is enough, it shouldn't be very useful to add extra rounds when training SuperPoint.

**Comment by Le0000000000n** [comment]
Thanks! I am thinking of training an extra round of super point on my own data to improve its performance on it.

**Comment by rpautrat** [comment]
Yes, fine-tuning on your own data is probably a good idea!

**Comment by Le0000000000n** [comment]
Thank you for your advice!

**Comment by UCRajkumar** [comment]
I'd like some clarification about this process. So, I train magic point on synthetic data, then export keypoints on mscoco. Then, train magicpoint from scratch on mscoco. From there, do I use this newly trained model to export keypoints again or use the same network from step 1? This is a silly question, but still want to make sure. 

Then once I export keypoints again, I train a new magicpoint model using the newly exported keypoints. Is that right?

**Comment by UCRajkumar** [comment]
So just to be clear, the second time I do step 2, I should be running the command `python export_detections.py configs/magic-point_coco_export.yaml magic_point_coco --pred_only --batch_size=5 --export_name=magic-point_coco-export1` instead of `python export_detections.py configs/magic-point_coco_export.yaml magic-point_synth --pred_only --batch_size=5 --export_name=magic-point_coco-export1
`

Is that right?

**Comment by rpautrat** [comment]
Yes. You can also change the name of the export if you don't want to override the previous exports and keep a record of the previous exports.
